cronjobs/add_authority_external_ref._automation.py

```diff
@@ -0,0 +1,183 @@
+    If the date of successful generating gnd_wiki (on first line) is older compare with the one on the web then:
+    a. Download the newer file from the web and put it into folder "/tmp"
+    b. The download file is a zip file, it needs to be extracted first
+    c. Get the information needed by gnd_wiki and put it into the file "/tmp/input_file_for_add_authority_external_ref.txt" using jq:
+        jq -c --stream '.' < authorities-gnd-person_lds.jsonld |grep -E 'https\:/\/d-nb\.info\/gnd\/|wikidata|wikipedia' > input_file_for_add_authority_external_ref.txt
+    d. Run "add_authority_external_ref" program. This program will check whether the lastest version date is newer compare to the lastest successful date generating gnd_wiki, if so then generate a new gnd_wiki.csv file. If it success generating gnd_wiki file then update the date on the config file.
+    e. Update the information in the config file
+
+"""
+
+import os
+import re
+import requests
+import util
+from urllib.request import urlopen
+import traceback
+from tqdm import tqdm
+import functools
+import shutil
+
+# this script needs:
+# 1. requests
+# 2. tqdm
+
+# Start global variable
+
+url = "https://data.dnb.de/opendata/"
+config_file = "/mnt/ZE020150/FID-Entwicklung/ub_tools/config_file_add_authority_ext_ref.cnf"
+share_folder = "/mnt/ZE020150/FID-Entwicklung/ub_tools/"
+# config_file = "/tmp/config_file_add_authority_ext_ref.cnf"
+# share_folder = "/tmp/"
+newer_file_name = ""
+gnd_wiki_file = "gnd_to_wiki.csv"
+current_file_date_integer = 0
+last_file_update_date_int = 0
+input_file_name_for_add_auth = "input_file_for_add_authority_external_ref.txt"
+
+
+# End global variable
+
+
+def DownloadTheFile(file_name):
+    url_path = f'{url}{file_name}'
+    target_file = f'/tmp/{file_name}'
+    dw = requests.get(url_path, stream=True)
+    if dw.status_code != 200:
+        dw.raise_for_status()  # Will only raise for 4xx codes, so...
+        raise RuntimeError(
+            f"Request to {url_path} returned status code {dw.status_code}")
+
+    file_size = int(dw.headers.get('Content-Length', 0))
+
+    desc = "(Unknown total file size)" if file_size == 0 else ""
+    dw.raw.read = functools.partial(
+        dw.raw.read, decode_content=True)  # Decompress if needed
+    with tqdm.wrapattr(dw.raw, "read", total=file_size, desc=desc) as r_raw:
+        f = open(target_file, "wb")
+        shutil.copyfileobj(r_raw, f)
+    return True
+
+
+def UpdateConfigFile(config_file, successful_generate_date):
+    add_auth_ext_ref_config_file = open(config_file, "w+")
+    add_auth_ext_ref_config_file.writelines(
+        [str(successful_generate_date), "\n"])
+
+
+def IsThereANewlyReleasedFile():
+    file_name = ""
+    global config_file
+    global current_file_date_integer
+    global last_file_update_date_int
+    global newer_file_name
```

Comment:
newer_file_name is declared not used in this function at all. Is it really necessary to declare a global variable for this, or can it be declared locally where it is needed?